helm-v3/007-security.md

```diff
@@ -13,4 +13,116 @@ Helm Classic). The following security goals are met by this proposal:
 - In-cluster CRDs can be secured with RBAC, too, providing access controls for
   releases.
 
+## Signed Charts by Default
+
+One of the weaknesses of the Helm 2 security model for chart signing is that
+it relied upon the user to explictly enable and configure signing. In Helm 3,
+signing will become the default.
+
+This proposal weakens some aspects of security (such as chain of trust), but in
+order to raise the overall security threshold.
+
+### Generating PGP Keys
+
+Upon the first run of `helm package`, a client will generate a PGP key if none
+is already present. This key pair will be stored in `$HELM_HOME`.
+
+Options will be present for importing a key from elsewhere, and the user will
```

Comment:
We should probably have a discussion around how we manage the key used to package the kubernetes/charts repo. cc @mattfarina @unguiculus @viglesiasce @scottrigby

Reply:
@technosophos have you considered how this would happen within a CI toolchain? Also, what about people using hardware such as a Yubikey?

Reply:
Basically, it will be possible to pre-seed a $HELM_HOME with the keyring, so CI systems will just use those keys. "Upon first run" is really intended to show what the use case is like for someone installing from scratch.

MFA is slightly more difficult, as the Go library wasn't very rich in this regard last time I looked. Depending on the state of that library when we get to implementation, we might either have to provide an option to shell out to another executable (like GnuPG) or basically tell people that they'll have to derive a key and use that without MFA.

Reply:
@technosophos The problem with the pre-seeded `$HELM_HOME` is where will the signing key be securely stored? For example, let's say I'm using Travis CI or CircleCI. How would I securely store the key between runs?

Reply:
The current method would still be usable, where you supply `--keyring` and `--key` as arguments to `helm package`. In that case, external tooling would be able to set up the keyring, and then Helm would just become a consumer of whatever it did.